components/validation/src/main/java/org/wso2/carbon/identity/x509Certificate/validation/CertificateValidationUtil.java

```diff
@@ -531,54 +537,156 @@ private static CACertificate resourceToCACertObject(Resource resource) throws Ce
         return new CACertificate(crlUrls, ocspUrls, x509Certificate);
     }
 
-    private static void addDefaultCACertificateInRegistry(Registry registry, String caCertRegPath,
-                                                          X509Certificate certificate,
-                                                          OMElement validatorChildElement) {
+    private static List<CACertificate> getCACertsFromConfigStore(String issuerDN) throws
+            ConfigurationManagementException, CertificateValidationException {
+
+        org.wso2.carbon.identity.configuration.mgt.core.model.Resource resource = CertValidationDataHolder.getInstance()
+                .getConfigurationManager()
+                .getResource(X509_CA_CERT_RESOURCE_TYPE, CERTS);
+
+        return getCertificateListFromResourceAndIssuerDN(resource, issuerDN);
+    }
+
+    private static List<CACertificate> getCertificateListFromResourceAndIssuerDN(
+            org.wso2.carbon.identity.configuration.mgt.core.model.Resource resource, String issuerDN)
+            throws CertificateValidationException {
+
+        List<CACertificate> certificateList = new ArrayList<>();
 
-        List<String> ocspUrls = new ArrayList<>();
-        List<String> crlUrls = new ArrayList<>();
         try {
-            if (!registry.resourceExists(caCertRegPath)) {
-                List<Validator> defaultValidatorConfig = getDefaultValidatorConfig(validatorChildElement);
-                boolean isSelfSignedCert = isSelfSignedCert(certificate);
-                for (Validator validator : defaultValidatorConfig) {
-                    if (validator.isEnabled()) {
-                        if (OCSP_VALIDATOR.equals(validator.getDisplayName()) && !isSelfSignedCert) {
-                            ocspUrls = getAIALocations(certificate);
-                        }
-                        else if (CRL_VALIDATOR.equals(validator.getDisplayName()) && !isSelfSignedCert) {
-                            crlUrls = getCRLUrls(certificate);
-                        }
-                    }
-                }
-                Resource resource = registry.newResource();
-                StringBuilder crlUrlReg = new StringBuilder();
-                if (CollectionUtils.isNotEmpty(crlUrls)) {
-                    for (String crlUrl : crlUrls) {
-                        crlUrlReg.append(crlUrl).append(CA_CERT_REG_CRL_OCSP_SEPERATOR);
-                    }
+            InputStream inputStream = CertValidationDataHolder.getInstance()
+                    .getConfigurationManager()
+                    .getFileById(X509_CA_CERT_RESOURCE_TYPE, CERTS, resource.getFiles().get(0).getId());
```

Comment:
Shall we handle the empty list in getFiles()

```
List<ResourceFile> resourceFiles = resource.getFiles();
if (resourceFiles == null || resourceFiles.isEmpty()) {
    log.warn("Resource files are empty for IssuerDN: " + issuerDN);
    return certificateList;
}

InputStream inputStream = CertValidationDataHolder.getInstance()
        .getConfigurationManager()
        .getFileById(X509_CA_CERT_RESOURCE_TYPE, CERTS, resourceFiles.get(0).getId());

if (inputStream == null) {
    log.warn("InputStream is null for the file in resource for IssuerDN: " + issuerDN);
    return certificateList;
}
```

Reply:
Fixed with https://github.com/wso2-extensions/identity-x509-commons/compare/7452398ba58b1246b4a707fdea8dfadf5e0376c0..29eee6b289366099e141bd2c7e3bfb17647fab4b